### PR TITLE
Add a new key `system` to the `hardware` spec

### PR DESCRIPTION
**Pull request by juk**

### CODE REVIEW

File: tmt/schemas/provision/hardware.yaml [hunk 1]
```diff
@@ -145,6 +145,32 @@ definitions:
...
+    additionalProperties: false
+
+    # enforce at least one property
+    minProperties: 1
+
+  # HW requirements: `system` block
+  systems:
+    type: array
+    items:
+      "$ref": "#/definitions/system"
```
Comment (psss) [inline tmt/schemas/provision/hardware.yaml:171]: I don't think we want to support mutliple `system` definitions. For `network` or `disks` it makes sense as we can require a guest with multiple network cards or multiple disks. But there would always we a single `vendor` or single number of numa nodes.

File: tmt/schemas/provision/hardware.yaml [hunk 2]
```diff
@@ -145,6 +145,32 @@ definitions:
...
+
+    properties:
+      vendor:
+        type: string
+
+      model:
+        type: string
+
+      numa-nodes:
+        type: string
```
Comment (psss) [inline tmt/schemas/provision/hardware.yaml:160]:
```suggestion
        type: integer
```

This should always be a number, right?

File: spec/hardware/system.fmf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+summary:
+    Select machine with given system attributes
+description:
+    Use the ``vendor`` key to choose system by specific vendor,
+    ``model`` to provide system's model name, ``numa-nodes``
+    to select systems with a given number of NUMA nodes
```
Comment (psss) [inline spec/hardware/system.fmf:6]:
```suggestion
    to select systems with a given number of NUMA nodes.
```

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for defining the `system` field. Looks ok, I would just remove possibility to provide multiple system definitions and require an `integer` for `numa-nodes`.

File: tmt/schemas/provision/hardware.yaml (hunk 1 above)
Comment (psss) [reply tmt/schemas/provision/hardware.yaml:171]: @happz, @thrix, @lukaszachy, what are your thoughts on this?

Comment (happz) [reply tmt/schemas/provision/hardware.yaml:171]: I'd say there might be a use case for getting a machine matching one of many possible vendors. I see no problem with that. On the other hand, I would stay away from `system` being a list, to keep things readable, and it's less common than NICs or storage devices. If there's a need to pick one of the multiple allowed system vendors, there's always `and` and `or`.

File: tmt/schemas/provision/hardware.yaml (hunk 2 above)
Comment (happz) [reply tmt/schemas/provision/hardware.yaml:160]: It should be both, an integer (`numa-nodes: 4`) and a string (`numa-nodes: ">= 4"`).

**Review by psss — APPROVED** [review]
Thanks for the updates. Now looks good. I've only added the example config to the test in 854f9281.

**Comment by psss** [comment]
Rebased and squashed, re-running tests.

**Comment by psss** [comment]
Interesting failure: Test `/tests/discover/dynamic-ref/fmf` fails with:

```
/plans/fmf/dynamic-ref/no-context
    discover
        how: fmf
        directory: /var/ARTIFACTS/work-basicf5rplsqr/plans/features/basic/discover/default-0/tests
        ref: fedora
        path: tests/discover/data
warn: /tests/discover1:test - True is not of type 'string'
    finish
        summary: 0 tasks completed
Traceback (most recent call last):
  File "/usr/bin/tmt", line 20, in <module>
    tmt.cli.main()
  ...
  File "/usr/lib/python3.11/site-packages/tmt/base.py", line 674, in _normalize_test
    return ShellScript(value)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/tmt/utils.py", line 309, in __init__
    self._script = textwrap.dedent(script)
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/textwrap.py", line 435, in dedent
    text = _whitespace_only_re.sub('', text)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'bool'
```

Which is caused by an outdated `fedora` branch in @juk's repository:

 * https://github.com/juk/tmt/blob/fedora/tests/discover/data/tests.fmf

Because it contains `test: true`.

@thrix, I had to dig through the [debug log](https://artifacts.dev.testing-farm.io/202d2396-cb9b-464a-882d-bd86e0907dd2/work-basicf5rplsqr/log.txt) because the [default log](https://artifacts.dev.testing-farm.io/202d2396-cb9b-464a-882d-bd86e0907dd2/) does not contain the test output, not sure why.

**Comment by psss** [comment]
The only failure is `/tests/discover/dynamic-ref/fmf` which is caused by an outdated content of the `fedora` branch in the forked repository. Everything else is green. Merging.

**Pull request merged**